themes-default/slim/src/components/edit-show.vue

```diff
@@ -0,0 +1,363 @@
+                { text: 'Skipped', value: 'Skipped' },
+                { text: 'Ignored', value: 'Ignored' }
+            ],
+            showLoaded: false,
+            saving: false
+        };
+    },
+    created() {
+        const { getShows } = this;
+        // Without getting any specific show data, we pick the show needed from the shows array.
+        getShows();
+    },
+    mounted() {
+        const { $store, indexer, id } = this;
+
+        // Let's tell the store which show we currently want as current.
+        $store.commit('currentShow', {
+            indexer,
+            id
+        });
+    },
+    methods: {
+        ...mapActions({
+            getShows: 'getShows'
+        }),
+        saveShow(subject) {
+            // We want to wait until the page has been fully loaded, before starting to save stuff.
+            if (!this.showLoaded) {
+                return;
+            }
+
+            if (!['show', 'all'].includes(subject)) {
+                return;
+            }
+
+            // Disable the save button until we're done.
+            this.saving = true;
+
+            const data = {
+                config: {
+                    aliases: this.show.config.aliases,
+                    defaultEpisodeStatus: this.show.config.defaultEpisodeStatus,
+                    dvdOrder: this.show.config.dvdOrder,
+                    seasonFolders: this.show.config.seasonFolders,
+                    anime: this.show.config.anime,
+                    scene: this.show.config.scene,
+                    sports: this.show.config.sports,
+                    paused: this.show.config.paused,
+                    location: this.show.config.location,
+                    airByDate: this.show.config.airByDate,
+                    subtitlesEnabled: this.show.config.subtitlesEnabled,
+                    release: {
+                        requiredWords: this.show.config.release.requiredWords,
+                        ignoredWords: this.show.config.release.ignoredWords,
+                        requiredWordsExclude: this.show.config.release.requiredWordsExclude,
+                        ignoredWordsExclude: this.show.config.release.ignoredWordsExclude
+                    },
+                    qualities: {
+                        preferred: this.show.config.qualities.preferred,
+                        allowed: this.show.config.qualities.allowed
+                    },
+                    airdateOffset: this.show.config.airdateOffset
+                },
+                language: this.show.language
+            };
+
+            if (data.config.anime) {
+                data.config.release.blacklist = this.show.config.release.blacklist;
+                data.config.release.whitelist = this.show.config.release.whitelist;
+            }
+
+            const { indexer, id, $store } = this;
+            $store.dispatch('setShow', { indexer, id, data, save: true }).then(() => {
```

Comment:
@sharkykh I'm not totally sure. Should we call an action in vuex, and have that 1. update store then 2. update the show using apiv2? Or update the show from here, like it's now?

Reply:
TL;DR: The action should call the patch and fetch the show again. I don't see a way around this at the moment.

I didn't look at the rest of the code, but since some of the patch data can be ignored by the API handler (invalid values and such), and whatever you send does not necessarily become the new show config, I'd say we can't update the store using the same data that we're sending, and we must fetch the show after the patch is complete.